client/blocks/app-banner/index.jsx

```diff
@@ -0,0 +1,148 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import classNames from 'classnames';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import Card from 'components/card';
+import { getActionLog } from 'state/ui/action-log/selectors';
+import {
+	getAppBannerData,
+	getPageType,
+	isDismissed,
+	getNewDismissTimes,
+	ALLOWED_PAGE_TYPES
+} from './utils';
+import {
+	findLast,
+	get,
+	identity,
+	includes,
+	noop
+} from 'lodash';
+import { isMobile } from 'lib/viewport';
+import { getUserSetting, isNotificationsOpen } from 'state/selectors';
+import { recordTracksEvent } from 'state/analytics/actions';
+import { ROUTE_SET } from 'state/action-types';
+import { saveUserSettings } from 'state/user-settings/actions';
+
+class AppBanner extends Component {
+	static propTypes = {
+		translate: PropTypes.func,
+		trackAppBannerImpression: PropTypes.func,
+		trackAppBannerDismiss: PropTypes.func,
+		trackAppBannerOpen: PropTypes.func,
+		saveDismissTime: PropTypes.func,
+		// connected
+		pageType: React.PropTypes.string,
+		dismissedUntil: React.PropTypes.object,
+	};
+
+	static defaultProps = {
+		translate: identity,
+		trackAppBannerImpression: noop,
+		trackAppBannerDismiss: noop,
+		trackAppBannerOpen: noop,
+		saveDismissTime: noop,
+		dismissedUntil: {}
+	};
+
+	state = {
+		isHidden: false,
+	};
+
+	isVisible() {
+		const { pageType, dismissedUntil } = this.props;
+
+		return isMobile() && ! isDismissed( dismissedUntil, pageType ) && ! this.state.isHidden;
```

Comment:
That sounds like a better alternative. I'll check out how it was done in get-apps and try to incorporate it here.